src/ext/default.c

```diff
@@ -209,6 +209,81 @@ static double BM25Scorer(const ScoringFunctionArgs *ctx, const RSIndexResult *r,
   return score;
 }
 
+/******************************************************************************************
+ *
+ * BM25 Scoring Functions - standard version according to https://en.wikipedia.org/wiki/Okapi_BM25
+ *
+ ******************************************************************************************/
+
+static double inline CalculateBM25Std(float b, float k1, double idf, double f, int doc_len,
+                                      double avg_doc_len, RSScoreExplain *scrExp, const char *term) {
+  double ret = idf * f * (k1 + 1) / (f + k1 * (1.0f - b + b * (float)doc_len/avg_doc_len));
+  EXPLAIN(scrExp,
+          "%s: (%.2f = IDF %.2f * (F %.2f * (k1 1.2 + 1)) / (F %.2f + k1 1.2 * (1 - b 0.5 + b 0.5 *"
+          " Doc Len %d / Average Doc Len %.2f)))",
+          term, ret, idf, f, f, doc_len, avg_doc_len);
+  return ret;
+}
+
+/* recursively calculate score for each token, summing up sub tokens */
+static double bm25StdRecursive(const ScoringFunctionArgs *ctx, const RSIndexResult *r,
+                            const RSDocumentMetadata *dmd, RSScoreExplain *scrExp) {
+  static const float b = 0.5f;
+  static const float k1 = 1.2f;
+  double f = (double)r->freq;
+  double ret = 0;
+  if (r->type == RSResultType_Term) {
+    // Compute IDF based on total number of docs in the index and the term's total frequency.
+    double idf = CalculateIDF_BM25(ctx->indexStats.numDocs, r->term.term ? r->term.term->numDocs : 0);
+    ret = CalculateBM25Std(b, k1, idf, f, dmd->len, ctx->indexStats.avgDocLen, scrExp,
+                           r->term.term->str);
+  } else if (r->type & (RSResultType_Intersection | RSResultType_Union | RSResultType_HybridMetric)) {
+    int numChildren = r->agg.numChildren;
+    if (!scrExp) {
+      for (int i = 0; i < numChildren; i++) {
+        ret += bm25StdRecursive(ctx, r->agg.children[i], dmd, NULL);
+      }
+    } else {
+      scrExp->numChildren = numChildren;
+      scrExp->children = rm_calloc(numChildren, sizeof(RSScoreExplain));
+      for (int i = 0; i < numChildren; i++) {
+        ret += bm25StdRecursive(ctx, r->agg.children[i], dmd, &scrExp->children[i]);
+      }
+      EXPLAIN(scrExp, "(Weight %.2f * children BM25 %.2f)", r->weight, ret);
```

Comment:
We probably want to show this is the stadndard one (or not the non-standard one)
```suggestion
      EXPLAIN(scrExp, "(Weight %.2f * children BM25STD %.2f)", r->weight, ret);
```

Reply:
Again, I rather not so it would be more readable...